sb_vision/tokens.py

```diff
@@ -58,107 +61,103 @@ def _get_pixel_centre(homography_matrix):
+                res_model=calibration.resolution,
+                res_this=image_size,
+            ),
         )
-        for i in range(4)
-    ]
-    return cartesian
-
-
-def _get_distance_for_family_day(pixel_corners, focal_length, token_height):
-    """HORRIBLE HACK."""
-    pixel_height = (
-        pixel_corners[1][1] -
-        pixel_corners[0][1] +
-        pixel_corners[2][1] -
-        pixel_corners[3][1]
-    ) / 4
-    return math.fabs(token_height * focal_length / pixel_height)
-
-
-def _get_distance(cartesian):
-    x = (
-        cartesian[0][0] +
-        cartesian[1][0] +
-        cartesian[2][0] +
-        cartesian[3][0]
-    ) / 4
-    y = (
-        cartesian[0][1] +
-        cartesian[1][1] +
-        cartesian[2][1] +
-        cartesian[3][1]
-    ) / 4
-    z = (
-        cartesian[0][2] +
-        cartesian[1][2] +
-        cartesian[2][2] +
-        cartesian[3][2]
-    ) / 4
-    return math.sqrt(x ** 2 + y ** 2 + z ** 2)
-
-
-def _cart_to_polar(cartesian_coord):
-    # TODO implement
-    # Don't bother making a struct for this, we will to send it over json in
-    # a sec. It's currently undefined what X, Y, and Z are. Go nuts.
-    rot_x, rot_y, rot_z, dist = 0, 0, 0, 0
-    return (rot_x, rot_y, rot_z), dist
-
-
-DEFAULT_TOKEN_SIZE = (0.25, 0.25)
+
+    return calibration
+
+
+def homography_matrix_to_distance_model_input_vector(homography_matrix):
+    """Convert a 3x3 homography matrix to a vector for distance models."""
+    flattened_vector = homography_matrix.ravel()
+    flattened_vector_as_row_matrix = np.array([flattened_vector])
+
+    quadratic_features = \
+        flattened_vector_as_row_matrix.T.dot(flattened_vector_as_row_matrix)
+    quadratic_features = quadratic_features.ravel()
+
+    all_features = np.hstack((
+        quadratic_features,
+        flattened_vector,
+    ))
+
+    return all_features
+
+
+def _apply_distance_model_component_to_homography_matrix(
+    model_component,
+    homography_matrix,
+):
+    flattened_homography_matrix = \
+        homography_matrix_to_distance_model_input_vector(homography_matrix)
+
+    biases = model_component['biases']
+    intercepts = model_component['intercept']
+    coefs = model_component['coefs']
+
+    return (
+        (biases + flattened_homography_matrix).dot(coefs) +
+        intercepts
+    )
+
+
+def _get_cartesian(
+    homography_matrix,
+    image_size,
+    distance_model,
+):
+    calibration = _get_distance_model(distance_model, image_size)
+
+    x = _apply_distance_model_component_to_homography_matrix(
+        calibration.x_model,
+        homography_matrix,
+    )
+    y = 0.0
+    z = _apply_distance_model_component_to_homography_matrix(
+        calibration.z_model,
+        homography_matrix,
+    )
+
+    return np.array([x, y, z])
 
 
 class Token:
     """Representation of the detection of one token."""
 
-    def __init__(self, id, size=DEFAULT_TOKEN_SIZE, certainty=0.0):
+    def __init__(self, id, certainty=0.0):
```

Comment:
Does the removal of token size mean that we only support a single size of token now? If not, where does the size of the token get taken into account?